File: CS/CustomService.cs
using DevExpress.Xpf.Grid;
using System;
using System.Windows;
using DevExpress.Mvvm.UI;

namespace DXGridThreads {

    public interface ICustomService {
        void BeginUpdate();
        void EndUpdate();
    }

    public class CustomService : ServiceBase, ICustomService {
        public static readonly DependencyProperty GridControlProperty =
            DependencyProperty.Register("GridControl", typeof(GridControl), typeof(CustomService), new PropertyMetadata(null));

        public GridControl GridControl {
            get { return (GridControl)GetValue(GridControlProperty); }
            set { SetValue(GridControlProperty, value); }
        }
        protected GridControl ActualGridControl { get { return GridControl != null ? GridControl : AssociatedObject as GridControl; } }

        public void BeginUpdate() {
            Dispatcher.Invoke(new Action(() => {
                if (ActualGridControl != null) {
                    ActualGridControl.BeginDataUpdate();
                }
            }));
        }

        public void EndUpdate() {
            Dispatcher.Invoke(new Action(() => {
                if (ActualGridControl != null) {
                    ActualGridControl.EndDataUpdate();
                }
            }));
        }
    }
}

File: VB/CustomService.vb
Imports DevExpress.Xpf.Grid
Imports DevExpress.Xpf.Mvvm.UI

Namespace DXGridThreads

    Public Interface ICustomService
        Sub BeginUpdate()
        Sub EndUpdate()
    End Interface

    Public Class CustomService
        Inherits ServiceBase
        Implements ICustomService

        Public Shared ReadOnly GridControlProperty As DependencyProperty = DependencyProperty.Register("GridControl", GetType(GridControl), GetType(CustomService), New PropertyMetadata(Nothing))

        Public Property GridControl() As GridControl
            Get
                Return CType(GetValue(GridControlProperty), GridControl)
            End Get
            Set(ByVal value As GridControl)
                SetValue(GridControlProperty, value)
            End Set
        End Property

        Protected ReadOnly Property ActualGridControl As GridControl
            Get
                If GridControl Is Nothing Then
                    Return TryCast(AssociatedObject, GridControl)
                End If
                Return GridControl
            End Get
        End Property

        Public Sub BeginUpdate() Implements ICustomService.BeginUpdate
            Dispatcher.Invoke(Sub()
                                  If ActualGridControl IsNot Nothing Then
                                      ActualGridControl.BeginDataUpdate()
                                  End If
                              End Sub)
        End Sub

        Public Sub EndUpdate() Implements ICustomService.EndUpdate
            Dispatcher.Invoke(Sub()
                                  If ActualGridControl IsNot Nothing Then
                                      ActualGridControl.EndDataUpdate()
                                  End If
                              End Sub)
        End Sub
    End Class
End Namespace
